main.tf

```diff
@@ -125,7 +125,9 @@ resource "aws_route_table" "public" {
   vpc_id = aws_vpc.default.id
   tags = merge(
     var.tags,
-    { "Name" = "${var.prepend_resource_type ? "route-table-" : ""}${var.name}-public" }
+    {
+      "Name" = var.shared_public_route_table ? "${var.prepend_resource_type ? "route-table-" : ""}${var.name}-public" : "${var.prepend_resource_type ? "route-table-" : ""}${var.name}-public-${local.az_ids[count.index]}"
+    }
```

Comment:
```suggestion
    {
        "Name" = "${var.prepend_resource_type ? "route-table-" : ""}${var.name}-public${var.shared_public_route_table ?  : "-${local.az_ids[count.index]}"}"
    }
```

Reply:
I didn't test/verify this, but I guess you get the point?

Reply:
will verify now looks good